packages/midi-input/src/MidiInput.ts

```diff
@@ -21,80 +24,103 @@ export class MidiInput implements IPlugin {
-          const node = storeState.nodes[input.targetNodeId]
-          const nodeVal = storeState.nodeValues[input.targetNodeId]
+      handleEachInput<typeof this.optionNodesConfig>(
+        storeState,
+        'midi',
+        ({ input, optionNodes: opts, targetNode, targetNodeValue }) => {
+          if (input.type !== 'midi') return
 
-          let value: NodeValue | null = null
+          if (
+            event.channel === opts.channel &&
+            event.note === opts.note &&
+            event.type === opts.type &&
+            event.value !== undefined
+          ) {
+            const sliderMin =
+              (storeState.nodeValues[`${input.targetNodeId}-sliderMin`] as number) ?? 0
+            const sliderMax =
+              (storeState.nodeValues[`${input.targetNodeId}-sliderMax`] as number) ?? 1
 
-          switch (node.valueType) {
-            case NodeTypes.Boolean:
-              switch (event.type) {
-                case MidiMessageType.NoteOn:
-                case MidiMessageType.NoteOff:
-                  value = !nodeVal
-                  break
-                default:
-                  value = event.value > 0
-                  break
-              }
-              break
-            case NodeTypes.Enum: {
-              switch (event.type) {
-                case MidiMessageType.NoteOn:
-                case MidiMessageType.NoteOff:
-                  value = getNextEnumValue(input.targetNodeId)(storeState)
-                  break
-                default: {
-                  const { options } = getParamConfig(input.targetNodeId)(
-                    storeState,
-                  ) as SketchConfigParamEnum
+            let value: NodeValue | null = null
 
-                  value = options[Math.floor((event.value / 127) * (options.length - 1))].value
-                  break
+            switch (targetNode.valueType) {
+              case 'boolean':
+                switch (event.type) {
+                  case MidiMessageType.NoteOn:
+                  case MidiMessageType.NoteOff:
+                    value = !targetNodeValue
+                    break
+                  default:
+                    value = event.value > 0
+                    break
                 }
+                break
+              case 'enum': {
+                switch (event.type) {
+                  case MidiMessageType.NoteOn:
+                  case MidiMessageType.NoteOff:
+                    value = getNextEnumValue(input.targetNodeId)(storeState)
+                    break
+                  default: {
+                    value =
+                      targetNode.options[
+                        Math.floor((event.value / 127) * (targetNode.options.length - 1))
+                      ].value
+                    break
+                  }
+                }
+                break
+              }
+              case 'number': {
+                value = (event.value / 127) * (sliderMax - sliderMin) + sliderMin
```

Comment:
I don't think it is appropriate to use slider min/max here. That is a top level param modifier, and I feel like inputs should be expected to just output a normalized value (which could potentially have it's own min/max like with lfo)